[#43] 추가된 단위 테스트에 따른 Service Layer 로직 수정

cyj199637 opened the pull request.

cyj199637 committed:
[#43] 추가된 단위 테스트에 따른 Service Layer 로직 수정

- 각 DTO 별로 필드 유효성 제약 조건 추가
- PostController 사용자가 존재하지 않는 게시물이나 존재하지 않는 사용자의 피드 조회를 요청하면 Not Found 응답을 반환하도록 로직 수정

cyj199637 committed:
Merge branch 'feature/29' into feature/43

f-lab-dev commented on src/main/java/me/liiot/snsserver/controller/PostController.java:
```diff
@@ -39,16 +41,24 @@
+            List<Post> posts = postService.getPostsByUser(userId);
 
-        return new ResponseEntity<>(posts, HttpStatus.OK);
+            return new ResponseEntity<>(posts, HttpStatus.OK);
```
이거는 팩토리메소드로 대체할 수 있을 것 같습니다~
`ResponseEntity.ok()`

cyj199637 committed:
Merge branch 'feature/29' into feature/43

cyj199637 committed:
[#43] 피드백 반영

- 일부 응답 객체 팩토리 메소드로 생성
- SessionLoginServiceTest의 추가된 테스트에 따른 SessionLoginService, UserController 로직 수정

cyj199637 committed:
Merge branch 'feature/29' into feature/43

f-lab-dev commented on src/main/java/me/liiot/snsserver/controller/UserController.java:
```diff
@@ -66,8 +63,13 @@
+
+            return RESPONSE_OK;
+        } catch (AlreadyLoginException e) {
+            return RESPONSE_BAD_REQUEST;
```
이건 사용자가 잘못된 요청을 보냈다기보다는 이미 로그인한 상태이기 때문에 에러가 발생하는거라 다른 코드를 쓰는게 나을 것 같아요~
그리고 꼭 에러를 보내줘야할지도 생각해봐야겠네요

f-lab-dev replied:
로그인한 유저가 다시 로그인을 시도한다고 해서 굳이 에러를 보내줄 필요는 없을 것 같아요~
이거는 상황에 따라 다르겠지만 정상 상황으로 처리해줘도 될 것 같아서 굳이 바꾸진 않더라도 한번 고민만 해보시면 될 것 같네요~

f-lab-dev commented:
테스트도 깨지는 것 같은데 이것도 확인 부탁드려요~

cyj199637 committed:
Merge branch 'feature/29' into feature/43

cyj199637 committed:
[#43] Jenkins Failed Test Code 수정

cyj199637 committed:
Merge branch 'feature/29' into feature/43

cyj199637 committed:
Merge branch 'feature/29' into feature/43

cyj199637 committed:
Merge branch 'feature/29' into feature/43
